src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2172,6 +2159,46 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                             // Parameter {0} must be declared with the '{1}' keyword
                             Error(diagnostics, ErrorCode.ERR_BadParamRef, lambdaParameterLocation, i + 1, delegateRefKind.ToParameterDisplayString());
                         }
+
+                        hasRefKindMismatch = true;
+                    }
+                }
+
+                if (!hasRefKindMismatch)
+                {
+                    for (int i = 0; i < anonymousFunction.ParameterCount; ++i)
```

Comment:
i don't get this why.  why do we need two loops?  we already have teh first loop.  and the first loop just sets a varaible to avoid diogn the second loop.

why not have one loop, and only do the type checks if there are no refkind issues?

Reply:
i don't get teh problem.

Reply:
Oh so then the filtering is per parameter, not per the entire parameter list of the lamdba, that greatly simplifies the processing involved.

Though one small problem I still have is when do we report the CS1661 (ERR_CantConvAnonMethParams) error, since we already report an error specific to the parameter itself, whether it's about the ref kind or the type itself?

Reply:
two options, and i leave it to the compiler to decide which is best:

1. do not report ERR_CantConvAnonMethParams if we've already reported a specific error about a specific parameter.
2. always report ERR_CantConvAnonMethParams, even if we've reported a specific error about a specific parameter.

either is fine with me.